java/src/main/native/CMakeLists.txt

```diff
@@ -52,7 +52,11 @@ message(VERBOSE "CUDF_JNI: Build with GPUDirect Storage support: ${USE_GDS}")
 message(VERBOSE "CUDF_JNI: Build with static Arrow library: ${CUDF_JNI_ARROW_STATIC}")
 
 set(CUDF_SOURCE_DIR "${PROJECT_SOURCE_DIR}/../../../../cpp")
-set(CUDF_CPP_BUILD_DIR "${CUDF_SOURCE_DIR}/build")
+if (DEFINED ENV{CUDF_CPP_BUILD_DIR}) 
+  set(CUDF_CPP_BUILD_DIR "$ENV{CUDF_CPP_BUILD_DIR}")
+else()
+  set(CUDF_CPP_BUILD_DIR "${CUDF_SOURCE_DIR}/build")
+endif()
```

Comment:
I have tried to push this change a few times, this works fine on my workstation. Can someone please test this patch to see if it doesn't break the flow that most devs use?

Reply:
You should be able to test the nightly Java build environment by following the instructions in [`java/ci/README.md`](https://github.com/rapidsai/cudf/blob/branch-21.08/java/ci/README.md)

Reply:
Looks like it passes 

```

2021-06-17 22:40:54 (3.44 MB/s) - 'gds-redistrib-0.95.1.tgz' saved [2143346/2143346]

Removing intermediate container 16dfd7215a18
 ---> bb7efd8d7571
Successfully built bb7efd8d7571
Successfully tagged cudf-build:11.2.2-devel-centos7

```

Reply:
so, the above was just building the docker, I followed the instructions and CI is building with my PR. Thank you for this @jlowe